### PR TITLE
Add scaffold for localizations (l10n) in Sharezone

### PR DESCRIPTION
**Pull request by flowhorn**
This pull request introduces a new package, sharezone_localizations, which provides a streamlined way of managing and accessing localized strings in the Sharezone-App using Flutter’s internationalization features. Below is an overview of the core additions:
	1.	New Package: sharezone_localizations
	•	README Documentation: Explains how to add, generate, and use translatable strings.
	•	Internationalization Setup: Based on Flutter’s gen-l10n tooling, enabling multiple locales and seamless code generation.
	2.	Locale Management
	•	`AppLocaleProvider`: Facilitates dynamic switching of app locales.
	3.	Usage Guidelines
	•	String Access: Via the context.sl extension, which simplifies referencing keys from the .arb files.
	•	Adding/Updating Strings: Detailed steps on modifying .arb files, including placeholder usage and recommended auto-translation tools.
	4.	Generating Localizations
	•	Flutter gen-l10n Command: Simple CLI approach to regenerate localizations after .arb updates.
	•	VS Code Task: Option to run a dedicated VS Code Task (“Generate l10n for sharezone_localizations”) for developers who prefer an IDE-based workflow.

Overall, this package consolidates translation logic, enhances maintainability by centralizing locale management, and simplifies how developers interact with localized strings.

image

Things, we need to do in future pull requests:
* https://github.com/SharezoneApp/sharezone-app/issues/1799
* Write tests for language page
* Store locale in user document (the backend have to know the language to send push notifications in the user's language)
* Replace hard-coded strings with l10n strings
* Remove old l10n files in `/app/l10n/`
* Document how to add new languages (copy `.arb` file, use [`arb_translate`](https://pub.dev/packages/arb_translate), makes native changes - `Info.plist` needs changed)
* Make sure that when system language is French, English is selected as default

## Summary by CodeRabbit

- **New Features**
	- Introduced a language selection feature in the settings.
	- Added localization support for German and English languages.
	- Implemented a `LanguagePage` for users to choose their preferred language.
	- Enhanced localization management with a new `FeatureFlagl10n` class.
	- Integrated dynamic locale adjustment based on user preferences.
	- Added a new `AppLocaleProvider` for managing application locale settings.
	- Established a structured localization framework with `SharezoneLocalizations`.
- **Bug Fixes**
	- Enhanced state management for localization features.
- **Documentation**
	- Added a comprehensive README for the `sharezone_localizations` package.
- **Chores**
	- Updated `.gitignore` to manage ignored files more effectively.

Part of #346

### CODE REVIEW

File: .vscode/tasks.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+            // This task generates the l10n files for the
+            // sharezone_localizations.
+            //
+            // FVM is required to run this task.
+            "label": "Generate l10n for sharezone_localizations",
+            "type": "shell",
+            // We also format the files after generating the l10n files because
+            // the generated files use a different formatting style than the
+            // rest of the project.
+            "command": "fvm flutter gen-l10n && fvm dart format .",
```
Comment (nilsreichardt) [inline .vscode/tasks.json:16]: Later, we can create a `sz` command for this: https://github.com/SharezoneApp/sharezone-app/issues/1799

File: app/lib/l10n/feature_flag_l10n.dart [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:sharezone/util/cache/streaming_key_value_store.dart';
+
+class FeatureFlagl10n extends ChangeNotifier {
```
Comment (nilsreichardt) [inline app/lib/l10n/feature_flag_l10n.dart:6]: Just a temporary class. Will be removed when we're done with translating.

File: app/lib/l10n/flutter_app_local_gateway.dart [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  @override
+  Stream<AppLocales> getLocale() {
+    final defaultValue = jsonEncode(featureFlagl10n.isl10nEnabled
+        ? AppLocales.system.toMap()
+        : AppLocales.en.toMap());
+    return keyValueStore
+        .getString('locale', defaultValue: defaultValue)
+        .map((event) => AppLocales.fromMap(jsonDecode(event)));
+  }
```
Comment (nilsreichardt) [inline app/lib/l10n/flutter_app_local_gateway.dart]: Later, we will stream the locale from Firebase. Will be done in a future pull request.

File: lib/sharezone_localizations/lib/src/app_locales.dart [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      return system;
+    }
+    return fromLanguageTag(map['languageTag']);
+  }
+
+  Map<String, dynamic> toMap() {
+    return {
+      'isSystem': isSystem(),
+      'languageTag': _toLanguageTag(),
+    };
```
Comment (nilsreichardt) [inline lib/sharezone_localizations/lib/src/app_locales.dart]: This is a proposal on how we could store it in the database. We need the `isSystem` to differentiate if the language is the system language or the selected language.

**Review by nilsreichardt — APPROVED** [review]
Thanks for starting to push this! ❤️

**Comment by nilsreichardt** [comment]
@Jonas-Sander You can review now 👍

**Comment by nilsreichardt** [comment]
@coderabbitai review

**Comment by nilsreichardt** [comment]
@coderabbitai review

File: app/lib/settings/src/subpages/language/language_page.dart [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+              ? null
+              : Text(locale.getTranslatedName(context)),
+          value: locale,
+          groupValue: localeProvider.locale,
+          onChanged: (value) {
+            localeProvider.locale = value!;
+          },
+        ),
+      ),
+    );
```
[the remark this answers is not in the document]
Comment (nilsreichardt) [reply app/lib/settings/src/subpages/language/language_page.dart:54]: `RadioListTile` already has these `Semantics` included. Not needed

**Review by Jonas-Sander — APPROVED** [review]
LGTM

**Pull request merged**